tracing-core/src/field.rs

```diff
@@ -165,10 +165,18 @@ pub struct FieldSet {
 
 /// A set of fields and values for a span.
 pub struct ValueSet<'a> {
-    values: &'a [(&'a Field, Option<&'a (dyn Value + 'a)>)],
+    values: Values<'a>,
     fields: &'a FieldSet,
 }
 
+enum Values<'a> {
+    /// A set of field-value pairs. Fields may be for the wrong field set, some
+    /// fields may be missing, and fields may be in any order.
+    Mixed(&'a [(&'a Field, Option<&'a (dyn Value + 'a)>)]),
```

Comment:
I wonder whether `Explicit` might be a better name than mixed. What do you think?